### DISCUSSION TITLE
"devmem2-like" utility in HSS' TinyCLI?

### DISCUSSION THREAD

**Discussion by haroldnelson-2021** [polarfire-soc/polarfire-soc-documentation#482] [Bare metal embedded software]
Is it feasible to add a "devmem2-like" utility to the HSS' TinyCLI so users can manually read-write their PF SOC fabric resources, assuming the PF SOC board has a UART connected to the PF SOC?  SmartDebug, of course, provides read-write access but this uses the JTAG interface and does not exercise the functional logic path.

**Comment by eppidei** [comment]
Hi @haroldnelson-2021 , there is debug menu in HSS. If you type "debug" you can look at the submenu. The hexdump utility should do what you are looking for.

**Comment by haroldnelson-2021** [comment]
I see "hexdump" in my TinyCLI.  Is there any help for the arguments and usage?  Thanks.  

Supported subcommands:
BEU HEALTHMON SM IPI CRC32 HEXDUMP OPENSBI SEG L2CACHE PERFCTR WDOG

**Reply by eppidei** [reply]
hi @haroldnelson-2021 , if you type debug hexdump with no args it should show usage info. You have to write the start address followed by length in hexadecimal format "0x<address>"
